airbyte-integrations/connectors/destination-iceberg-v2/src/main/kotlin/io/airbyte/integrations/destination/iceberg/v2/io/IcebergUtil.kt

```diff
@@ -0,0 +1,175 @@
+     * @return The Iceberg [Table], created if it does not yet exist.
+     */
+    fun createTable(
+        tableIdentifier: TableIdentifier,
+        catalog: Catalog,
+        schema: Schema,
+        properties: Map<String, String>
+    ): Table {
+        if (
+            catalog is SupportsNamespaces && !catalog.namespaceExists(tableIdentifier.namespace())
+        ) {
+            catalog.createNamespace(tableIdentifier.namespace())
+            logger.info { "Created namespace '${tableIdentifier.namespace()}'." }
+        }
+
+        return if (!catalog.tableExists(tableIdentifier)) {
+            logger.info { "Creating Iceberg table '$tableIdentifier'...." }
+            catalog
+                .buildTable(tableIdentifier, schema)
+                .withProperties(properties)
+                .withProperty(DEFAULT_FILE_FORMAT, FileFormat.PARQUET.name.lowercase())
+                .withSortOrder(getSortOrder(schema = schema))
+                .create()
+        } else {
+            logger.info { "Loading Iceberg table $tableIdentifier ..." }
+            catalog.loadTable(tableIdentifier)
+        }
+    }
+
+    /**
+     * Converts an Airbyte [DestinationRecord] into an Iceberg [Record]. The converted record will
+     * be wrapped to include [Operation] information, which is used by the writer to determine how
+     * to write the data to the underlying Iceberg files.
+     *
+     * @param record The Airbyte [DestinationRecord] record to be converted for writing by Iceberg.
+     * @param stream The Airbyte [DestinationStream] that contains information about the stream.
+     * @param tableSchema The Iceberg [Table] [Schema].
+     * @param pipeline The [MapperPipeline] used to convert the Airbyte record to an Iceberg record.
+     * @return An Iceberg [Record] representation of the Airbyte [DestinationRecord].
+     */
+    fun toRecord(
+        record: DestinationRecord,
+        stream: DestinationStream,
+        tableSchema: Schema,
+        pipeline: MapperPipeline
+    ): Record {
+        val dataMapped =
+            pipeline
+                .map(record.data, record.meta?.changes)
+                .withAirbyteMeta(stream, record.emittedAtMs, true)
+        // TODO figure out how to detect the actual operation value
+        return RecordWrapper(
+            delegate = dataMapped.toIcebergRecord(tableSchema),
+            operation = Operation.INSERT
+        )
+    }
+
+    /**
+     * Creates the Iceberg [Catalog] configuration properties from the destination's configuration.
+     *
+     * @param icebergConfiguration The destination's configuration
+     * @return The Iceberg [Catalog] configuration properties.
+     */
+    fun toCatalogProperties(icebergConfiguration: IcebergV2Configuration): Map<String, String> {
+        return mutableMapOf(
+                // TODO make configurable?
+                ICEBERG_CATALOG_TYPE to ICEBERG_CATALOG_TYPE_NESSIE,
```

Comment:
fyi was talking to alex earlier today - we'll want to add glue support to this connector before releasing it globally, so this should be configurable somehow